mobly/base_test.py

```diff
@@ -862,6 +862,11 @@ def generate_tests(self, test_logic, name_func, arg_sets, uid_func=None):
         raise Error('%s Test name "%s" already exists, cannot be duplicated!' %
                     (root_msg, test_name))
       test_func = functools.partial(test_logic, *args)
+      for attr_name in (
```

Comment:
we are essentially applying the decoration attrs to the generated func right?
can you add an inline comment explaining what this is doing so it's easier to understand for ppl without the context?

Reply:
Comments added! Thanks!